sklearn/ensemble/iforest.py

```diff
@@ -437,29 +440,22 @@ def _average_path_length(n_samples_leaf):
     average_path_length : array, same shape as n_samples_leaf
 
     """
-    if isinstance(n_samples_leaf, INTEGER_TYPES):
```

Comment:
I don't see how this relates to the topic of the current PR. It seems cosmetic, but I've not really understood why you'r removing this logic.

Reply:
this PR simplifies this function by avoiding having 2D arrays as inputs (just 1D arrays instead). We can thus include the case where the input is a float by considering it as a 1d array (before the change, we could have consider a float input as a 2d array I guess, but it was probably a bit too much of a hack).